Sources/NIOResumableUpload/HTTPResumableUploadContext.swift

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOConcurrencyHelpers
+import NIOCore
+
+/// `HTTPResumableUploadContext` manages ongoing uploads.
+public final class HTTPResumableUploadContext {
```

Comment:
Hey I'll be pitching in for Guoye to address feedback. I think `HTTPResumableUploadContext` is public because it provides a nice way for developers to configure unique or optional behaviors of their resumable uploads. In the future, the protocol draft may contain optional features such as progress reporting, and if we support them, this context could be the place to opt in or out (instead of passing more parameters to the `handler`). Guoye may have other considerations, too--I can ask when he's back from vacation.